### PR TITLE
Implement tag support in SELECT statements

### LINKED ISSUE #1989: implement `SELECT tagName FROM m`
Implement selecting tag values to be displayed in the results.

### PR DESCRIPTION
**Pull request by toddboom**
Fixes #1989 and #2499.

### CODE REVIEW

File: cmd/influxd/server_integration_test.go [hunk 1]
```diff
@@ -460,10 +460,23 @@ func runTestsData(t *testing.T, testName string, nodes Cluster, database, retent
...
+			query:    `SELECT host FROM "%DB%"."%RP%".gpu`,
+			expected: `{"results":[{"error":"select statement must include at least one field"}]}`,
+		},
 		{
-			name:     "single point, select with now(), two queries",
-			query:    `SELECT * FROM "%DB%"."%RP%".cpu WHERE time < now();SELECT * FROM "%DB%"."%RP%".cpu WHERE time < now()`,
-			expected: `{"results":[{"series":[{"name":"cpu","columns":["time","value"],"values":[["2015-02-28T01:03:36.703820946Z",100]]}]},{"series":[{"name":"cpu","columns":["time","value"],"values":[["2015-02-28T01:03:36.703820946Z",100]]}]}]}`,
+			name:     "selecting a tag and a field",
+			query:    `SELECT host, value FROM "%DB%"."%RP%".gpu`,
+			expected: `{"results":[{"series":[{"name":"gpu","tags":{"host":"server01"},"columns":["time","value"],"values":[["2015-02-28T01:03:36.703820946Z",100]]}]}]}`,
```
Comment (pauldix) [inline cmd/influxd/server_integration_test.go]: Can you update this test so it has two data points from different hosts?

**Comment by pauldix** [comment]
+1

File: influxql/engine.go [hunk 2]
```diff
@@ -711,13 +711,25 @@ func (m *MapReduceJob) processRawResults(values []*rawQueryMapOutput) *Row {
 		selectNames = append([]string{"time"}, selectNames...)
 	}
 
-	// if they've selected only a single value we have to handle things a little differently
-	singleValue := len(selectNames) == SelectColumnCountWithOneValue
+	// since selectNames can contain tags, we need to strip them out
+	selectFields := make([]string, 0, len(selectNames))
+
+	for _, n := range selectNames {
```
Comment (corylanou) [inline influxql/engine.go]: I think you can shorten this to:

```go
 for _, n := range selectNames {            
   if _, found := m.TagSet.Tags[n]; !found {
     selectFields = append(selectFields, n) 
   }                                        
 }                                          
```

Comment (toddboom) [inline influxql/engine.go]: yes, plz! doing it now.

**Comment by corylanou** [comment]
Minor feedback, looks great.  +1 :shipit:
